### ISSUE TITLE
ably-go: add logging to rest & realtime clients

### ISSUE THREAD

**Issue by rjeczalik** [ably/ably-go#38]
Currently logging is implemented in [logger.go](https://github.com/ably/ably-go/blob/master/ably/logger.go) but not used everywhere.

**Comment by mattheworiordan** [comment]
@rjeczalik is this still an issue?

**Comment by rjeczalik** [comment]
@mattheworiordan Yes, still an issue. Current state is the logger value is properly propagated throughout all types in the library. Missing piece is to do the actual logging.
